### DISCUSSION TITLE
Arrow padding and placement "right-start"

### DISCUSSION THREAD

**Discussion by PavelZhuravlev** [floating-ui/floating-ui#1390] [Q&A] [answered]
Hello!
Thank you for awesome library!

Can you help me understand, is there any way I can compensate arrow padding and keep arrow in the center of reference when placement is set to `*-start` or `*-end`

Current behavior with combination of padding and placement `right-start"
![Снимок экрана 2021-10-21 в 04 51 43](https://user-images.githubusercontent.com/14184094/138197593-d38b4f82-2207-4dff-8214-4bab3425c49a.png)

Behavior I need to achieve:
![Снимок экрана 2021-10-21 в 04 52 20](https://user-images.githubusercontent.com/14184094/138197673-92f99f14-c426-4443-b06b-35377c50fd11.png)

Demo: https://codesandbox.io/s/cool-lumiere-j0x00?file=/index.html

Thanks!

**Comment by atomiks** [comment, marked as answer]
`offset: [-16, 16]` appears to work

**Reply by PavelZhuravlev** [reply]
@atomiks 
Thank you for the answer
but actually negative offset doesn't work ,
 because it breaks the `right-end` placement

**Reply by PavelZhuravlev** [reply]
![Снимок экрана 2021-10-21 в 05 16 11](https://user-images.githubusercontent.com/14184094/138199606-6097c42c-334b-4a95-b032-408061973375.png)

**Reply by atomiks** [reply]
@PavelZhuravlev you need to check the placement: `offset: ({ placement }) => [placement.endsWith('-end') ? 16 : -16, 15]` though it's an API flaw, see https://github.com/popperjs/popper-core/issues/1170

**Reply by PavelZhuravlev** [reply]
@atomiks  Thank you!
